apps/backend/src/app/api/admin/semesters/[id]/route.ts

```diff
@@ -14,13 +20,25 @@ class SemesterRouteWrapper {
    * @returns No content status code
    */
   @Security("jwt", ["admin"])
-  static async DELETE(_req: NextRequest, { params }: { params: Promise<{ id: string }> }) {
+  static async DELETE(req: NextRequest, { params }: { params: Promise<{ id: string }> }) {
+    const { id } = await params
+
+    const deletedRelatedDocs = req.nextUrl.searchParams.get("deleteRelatedDocs") !== "false"
+    const cascadeTransactionId = await getTransactionId(!!deletedRelatedDocs)
+
     try {
-      const { id } = await params
       const semesterDataService = new SemesterDataService()
-      await semesterDataService.deleteSemester(id)
+      if (deletedRelatedDocs) {
+        await semesterDataService.deleteRelatedDocsForSemester(id, cascadeTransactionId)
+      }
+      await semesterDataService.deleteSemester(id, cascadeTransactionId)
+
+      await commitCascadeTransaction(cascadeTransactionId)
+
       return new NextResponse(null, { status: StatusCodes.NO_CONTENT })
     } catch (error) {
+      await rollbackCascadeTransaction(cascadeTransactionId)
+
       if (error instanceof NotFound) {
         return NextResponse.json({ error: "Semester not found" }, { status: StatusCodes.NOT_FOUND })
       }
```

Comment:
just realised this route is missing the standard `console.error(error)` after this line

```suggestion
      if (error instanceof NotFound) {
        return NextResponse.json({ error: "Semester not found" }, { status: StatusCodes.NOT_FOUND })
      }
      console.error(error)
```